### PR TITLE
julia: test build

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Trying to see if the linux build still fails after the gcc 12 migration.

**Comment by carlocab** [comment]
Related: https://github.com/Homebrew/homebrew-core/pull/107388#issuecomment-1206655030

**Comment by carlocab** [comment]
Same error. The failure doesn't seem to be coming from any particular test case in upstream's tests but from setup code. In particular, it's not happy about a float (`1.526e-5`) being parsed as an `Int64`.

**Comment by cho-m** [comment]
Could take a look at Arch's changes/patches for GCC 12: https://github.com/archlinux/svntogit-community/commit/bee1243b4ec66da31097f84600b37451435cfb1e

File: Formula/julia.rb [hunk 1]
```diff
@@ -45,6 +45,12 @@ class Julia < Formula
       url "https://github.com/JuliaLang/julia/commit/4d7fc8465ed9eb820893235a6ff3d40274b643a7.patch?full_index=1"
       sha256 "3a34a2cd553929c2aee74aba04c8e42ccb896f9d491fb677537cd4bca9ba7caa"
     end
+
+    # Fixes for gcc12 https://github.com/Homebrew/homebrew-core/pull/107875#issuecomment-1214216233
+    patch do
+      url "https://github.com/archlinux/svntogit-community/commit/bee1243b4ec66da31097f84600b37451435cfb1e.patch?full_index=1"
+      sha256 "8a048507785540f83cff2d1d4f26cae8544025f17b64a8c3be1368c2b22fe345"
```
Comment (cho-m) [inline Formula/julia.rb:52]: Comment was actually commit for Arch's package update. Need to grab patch file from that or upstream:
```suggestion
      url "https://raw.githubusercontent.com/archlinux/svntogit-community/bee1243b4ec66da31097f84600b37451435cfb1e/trunk/julia-gcc-12.patch"
      sha256 "b903859c262048057f873e4a39c4837382896d3059860465443823133a5a59da"
```

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
https://github.com/Homebrew/homebrew-core/actions/runs/2871959481

**Pull request merged**